ecl/eclagent/eclgraph.cpp

```diff
@@ -853,9 +853,16 @@ void EclSubGraph::updateProgress()
 {
     if (!isChildGraph && agent->queryRemoteWorkunit())
     {
-        Owned<IWUGraphStats> progress = parent.updateStats(queryStatisticsComponentType(), queryStatisticsComponentName(), id);
+        Owned<IWUGraphStats> progress = parent.updateStats(queryStatisticsComponentType(), queryStatisticsComponentName(), agent->getWorkflowId(), id);
         IStatisticGatherer & stats = progress->queryStatsBuilder();
         updateProgress(stats);
+
+        StringBuffer subgraphid;
+        subgraphid.append(parent.queryGraphName()).append(":").append(SubGraphScopePrefix).append(id);
+        if (startGraphTime)
```

Comment:
Better to lock the wu once, rather than once each call.  Does locking the workunit introduce any potential problems?

Reply:
Code modified so that the wu is locked once before the 2 updates.  I haven't experienced any problems with any wu failing because of this.  Does hthor executed a large number of graphs concurrently?